src/mqt/bench/utils.py

```diff
@@ -342,3 +358,74 @@ def postprocess_single_oqc_file(filename: str):
 
 def create_zip_file():
     return subprocess.call(f"zip -rj {get_zip_file_path()} {get_qasm_output_path()}", shell=True)
+
+
+def calc_qubit_index(qargs: list[Qubit], qregs: list[QuantumRegister], index: int) -> int:
+    offset = 0
+    for reg in qregs:
+        if qargs[index] not in reg:
+            offset += reg.size
+        else:
+            qubit_index = offset + reg.index(qargs[index])
+            return qubit_index
+    error_msg = f"Global qubit index for local qubit {index} index not found."
+    raise ValueError(error_msg)
+
+
+def calc_supermarq_features(
+    qc: QuantumCircuit,
+) -> tuple[float, float, float, float, float]:
+    qc.remove_final_measurements(inplace=True)
+    qc = RemoveBarriers()(qc)
+    connectivity_collection: list[list[int]] = []
+    liveness_A_matrix = 0
+    for _ in range(qc.num_qubits):
+        connectivity_collection.append([])
+
+    for _, qargs, _ in qc.data:
+        liveness_A_matrix += len(qargs)
+        first_qubit = calc_qubit_index(qargs, qc.qregs, 0)
+        all_indices = [first_qubit]
+        if len(qargs) == 2:
+            second_qubit = calc_qubit_index(qargs, qc.qregs, 1)
+            all_indices.append(second_qubit)
+        for qubit_index in all_indices:
+            to_be_added_entries = all_indices.copy()
+            to_be_added_entries.remove(int(qubit_index))
+            connectivity_collection[int(qubit_index)].extend(to_be_added_entries)
+
+    connectivity: list[Any] = []
```

Comment:
Needs a proper type ;o)
Shouldn't be too hard to infer.

Reply:
Yes, was quite simple. Additionally, I improved the `connectivity` initialization routine below.Yes, was quite simple. Additionally, I improved the `connectivity` initialization routine below.